### PR TITLE
Place build_runner.zig in cache directory

### LINKED ISSUE #634: Don't drop build_runner.zig next to executable
### Zig Version

0.10.0-dev.3685+dae7aeb33

### Zig Language Server Version

38210d3e9183a77dfe10e0d0f4a79d2999ce3cf7

### Steps to Reproduce

Start `zls` in a project. If the `zls` binary is installed in a read-only directory, such as `/usr/bin` this will crash with a permission denied error.

### Expected Behavior

`zls` doesn't try to drop any files next to its executable, and instead uses some alternative path such as `zig-cache` or `/tmp`.

### Actual Behavior

`zls` tries to create a `build_runner.zig` file next to its executable. This often fails, for example when zls is installed in `/usr/bin` (tries to create `/usr/bin/build_runner.zig`), and needlessly pollutes the filesystem.

This was introduced in `38210d3e9183a77dfe10e0d0f4a79d2999ce3cf7`.

Once we agree on a solution, I'd very much like to implement it. Cheers!

### PR DESCRIPTION
**Pull request by Techatrix**
fixes #634

### CODE REVIEW

File: src/Config.zig [hunk 1]
```diff
@@ -202,10 +202,13 @@ pub fn configChanged(config: *Config, allocator: std.mem.Allocator, builtin_crea
         config.builtin_path = try std.fs.path.join(allocator, &.{ builtin_creation_dir.?, "builtin.zig" });
     }
 
+    const cache_dir_path = (try known_folders.getPath(allocator, .cache)) orelse {
```
Comment (LordMZTE) [inline src/Config.zig:205]: You should append `zls` to this. It's bad practice to drop stuff straight in `~/.cache`.

**Review by SuperAuguste — APPROVED** [review]
LGTM!

File: src/Config.zig [hunk 2]
```diff
@@ -214,10 +217,6 @@ pub fn configChanged(config: *Config, allocator: std.mem.Allocator, builtin_crea
     }
 
     if (null == config.global_cache_path) {
```
Comment (LordMZTE) [inline src/Config.zig:219]: Might be best to just move this block above the one creating the build runner file instead of using `cache_dir_path`

**Review by LordMZTE — CHANGES_REQUESTED** [review]
Sniped my PR :P

**Pull request merged**